tsl/src/remote/connection.c

```diff
@@ -455,6 +455,11 @@ extract_connection_options(List *defelems, const char **keywords, const char **v
 		}
 	}
 
+	/* If a user was not found in the connection options, assume the current
+	 * user */
+	if (NULL == *user)
+		*user = GetUserNameFromId(GetUserId(), true);
+
 	Assert(*user != NULL);
```

Comment:
How this assert worked before?

Reply:
Not sure what you mean, but this is where the crash happened without my changes when assert was enabled. Without asserts (in release builds) it crashed later when it tried to md5-hash the user name to create a directory for a user SSL certs.

Reply:
Oh okey, so it worked and crashed at that point. Makes sense then